### PR TITLE
Magic Login: Force a full page reload when clicking "Back to WordPress.com"

### PR DESCRIPTION
**Pull request by jblz**
Thought to have been squashed in #15735, wpcalypso was still exhibiting the previous behavior:

On the “Check your email” screen and “Expired/Invalid link” screen the button “Back to WordPress.com” doesn’t appear to do anything despite the browser navigating to `/`.

This adds an explicit window.location call via a "pseudo" action function to trigger the route update & an explicit action URL of `https://wordpress.com` in each usage.

It also converges the CTA strings to `Back to WordPress.com` -- we don't need both that & `Return to WordPress.com`.

## To Test

Using an Incognito or logged-out window:

### Check the button on the "Check Your Email" page:
* Browse to: http://calypso.localhost:3000/log-in/link
* Issue in the console: `dispatch( { type: 'MAGIC_LOGIN_SHOW_CHECK_YOUR_EMAIL_PAGE' } )`
* Click the `Back to WordPress.com` button
* The route should change to `/` & the screen should change (in dev, you'll land at `/devdocs/start` since we redirect, but that's ok)

### Check the button on the "Link is Expired... page"
* Browse to: http://calypso.localhost:3000/log-in/link/use?client_id=.&email=example@example.com&token=.&tt=.
* Issue in the console: `dispatch( { type: 'MAGIC_LOGIN_SHOW_LINK_EXPIRED' } )`
* Click the `Return to WordPress.com` button
* The route should change to `/` & the screen should change (in dev, you'll land at `/devdocs/start` since we redirect, but that's ok)

### CODE REVIEW

File: client/login/magic-login/emailed-login-link-expired.jsx [hunk 1]
```diff
@@ -37,11 +37,9 @@ class EmailedLoginLinkExpired extends React.Component {
 					replaceCurrentLocation={ true }
 				/>
 				<EmptyContent
-					action={ translate( 'Return to WordPress.com' ) }
-					actionCallback={ function() {
-						page( '/' );
-					} }
-					actionURL={ '/' }
+					action={ translate( 'Back to WordPress.com' ) }
+					actionCallback={ goBackToWordPressDotCom }
```
Comment (bisko) [inline client/login/magic-login/emailed-login-link-expired.jsx]: Looking at the code for the `EmptyContent` component, `actionCallback` is not required here. The `<a>` tag will properly get its `href` value set to the absolute URL we pass below in `actionURL`. 
This will make it open the URL without using the `page()` framework.

File: client/login/magic-login/emailed-login-link-expired.jsx [hunk 2]
```diff
@@ -37,11 +37,9 @@ class EmailedLoginLinkExpired extends React.Component {
...
+					actionURL={ 'https://wordpress.com' }
```
Comment (bisko) [inline client/login/magic-login/emailed-login-link-expired.jsx]: I think it's best to pass static strings directly as attribute values like:

```
actionURL="https://wordpress.com"
```

File: client/login/magic-login/emailed-login-link-successfully.jsx [hunk 3]
```diff
@@ -45,10 +45,8 @@ class EmailedLoginLinkSuccessfully extends React.Component {
 				/>
 				<EmptyContent
 					action={ translate( 'Back to WordPress.com' ) }
-					actionCallback={ function() {
-						page( '/' );
-					} }
-					actionURL={ '/' }
+					actionCallback={ goBackToWordPressDotCom }
```
Comment (bisko) [inline client/login/magic-login/emailed-login-link-successfully.jsx]: Same as my comment on `emailed-login-link-expired.jsx` above.

File: client/state/login/magic-login/actions.js [hunk 4]
```diff
@@ -111,3 +111,7 @@ export const fetchMagicLoginAuthenticate = ( email, token, tt ) => dispatch => {
 			} );
 		} );
 };
+
+export function goBackToWordPressDotCom() {
```
Comment (bisko) [inline client/state/login/magic-login/actions.js]: If we decide to omit `actionCallback` above, this will not be needed. 

I'm not sure if `state` is the best place to put this in, since this doesn't directly work with the Redux state.

File: client/login/magic-login/emailed-login-link-expired.jsx (hunk 1 above)
Comment (jblz) [inline client/login/magic-login/emailed-login-link-expired.jsx]: @bisko -- That's what this was originally doing, just linking to `/` via the `actionURL` prop.

Check out the description of the problem this is solving (in this PR & in #15735).  Some routing magic somewhere was intercepting the page meaning that the route was changing to `/`, but the app state was not changing.

I think this is due to us not having a "real" root section in Calypso.  This should work around it.

File: client/state/login/magic-login/actions.js (hunk 4 above)
Comment (jblz) [inline client/state/login/magic-login/actions.js]: > I'm not sure if state is the best place to put this in, since this doesn't directly work with the Redux state.

The decision to put it in the actions file was based on @mcsf's suggestion:

> don’t create the function in `render`, maybe make it a pseudo-action so that you can reuse it in both components

Happy to move it if you can think of a better place, but technically, route changes are actions affecting app state, if you want to get technical :P

File: client/login/magic-login/emailed-login-link-expired.jsx (hunk 1 above)
Comment (bisko) [inline client/login/magic-login/emailed-login-link-expired.jsx]: Yes, if we want to keep it dynamic, then the action should be OK to keep here.

File: client/state/login/magic-login/actions.js (hunk 4 above)
Comment (bisko) [inline client/state/login/magic-login/actions.js]: > but technically, route changes are actions affecting app state, if you want to get technical :P

True that :) 

As it makes more sense to keep the action for now, let's leave it here. If we need to move it, we can do it in a separate PR.
